pkg/pool/pool.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pool
+
+import (
+	"sync"
+)
+
+type impl struct {
+	wg     sync.WaitGroup
+	workCh chan func() error
+	errCh  chan error
+}
+
+// impl implements Interface
+var _ Interface = (*impl)(nil)
+
+// BufferSize is the number of work items or errors that we
+// can queue up before calls to Go will block, or work will
+// block until Wait is called.
+const BufferSize = 50
+
+// New creates a fresh worker pool with the specified size.
+func New(workers int) Interface {
+	i := &impl{
+		workCh: make(chan func() error, BufferSize),
+		errCh:  make(chan error, BufferSize),
+	}
+
+	// Start a go routine for each worker, which:
+	// 1. reads off of the work channel,
+	// 2. (optionally) sends errors on the error channel,
+	// 3. marks work as done in our sync.WaitGroup
+	for idx := 0; idx < workers; idx++ {
+		go func() {
+			for work := range i.workCh {
+				if err := work(); err != nil {
+					i.errCh <- err
+				}
+				i.wg.Done()
+			}
+		}()
+	}
+
+	return i
+}
+
+// Go implements Interface
```

Comment:
This sentence is so recursive 😉 
```suggestion
// Go implements Interface.
```

Reply:
Done.